File: ios/Features/Config/ConfigModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ConfigModule, NSObject)

RCT_EXTERN__BLOCKING_SYNCHRONOUS_METHOD(initializeHereSDK:(NSString *)accessKeyID
                                        withAccessKeySecret:(NSString *)accessKeySecret)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/Features/Config/ConfigModule.swift
import heresdk

@objc(ConfigModule)
class ConfigModule: NSObject {
    
    @objc(initializeHereSDK:withAccessKeySecret:)
    func initializeHereSDK(accessKeyID: String, accessKeySecret: String) -> String {
        let options = SDKOptions(
            accessKeyId: accessKeyID,
            accessKeySecret: accessKeySecret)
        do {
            try SDKNativeEngine.makeSharedInstance(options: options)
            return "SDKNativeEngine started"
        } catch _ {
            return "SDKNativeEngine errored"
        }
    }
}
